content/guides/references/migration-guide.md

```diff
@@ -2,7 +2,361 @@
+removed. Use the [`cy.session()`](/api/commands/session) command to preserve
+cookies (and local and session storage) between tests.
+
+```diff
+describe('Dashboard', () => {
+  beforeEach(() => {
+-    cy.login()
+-    Cypress.Cookies.preserveOnce('session_id', 'remember_token')
++    cy.session('unique_identifier', cy.login, {
++       validate () {
++        cy.getCookies().should('have.length', 2)
++       },
++       cacheAcrossSpecs: true
++    })
+  })
+```
+
+#### `cy.server()`, `cy.route()` and `Cypress.Server.defaults`
+
+The` cy.server()` and` cy.route()` commands and the `Cypress.server.defaults`
+API has been removed. Use the [`cy.intercept()`](/api/commands/intercept)
+command instead.
+
+```diff
+  it('can encode + decode headers', () => {
+-   Cypress.Server.defaults({
+-     delay: 500,
+-     method: 'GET',
+-   })
+-   cy.server()
+-   cy.route(/api/, () => {
+-      return {
+-        'test': 'We’ll',
+-      }
+-    }).as('getApi')
++   cy.intercept('GET', /api/, (req) => {
++      req.on('response', (res) => {
++        res.setDelay(500)
++      })
++      req.body.'test': 'We’ll'
++    }).as('getApi')
+    cy.visit('/index.html')
+    cy.window().then((win) => {
+      const xhr = new win.XMLHttpRequest
+      xhr.open('GET', '/api/v1/foo/bar?a=42')
+      xhr.send()
+    })
+
+    cy.wait('@getApi')
+-   .its('url').should('include', 'api/v1')
++   .its('request.url').should('include', 'api/v1')
+  })
+```
+
+#### `.invoke()`
+
+[`.invoke()`](/api/commands/invoke) now throws an error if the function returns
+a promise. If you wish to call a method that returns a promise and wait for it
+to resolve, use [`.then()`](/api/commands/then) instead of `.invoke()`.
+
+```diff
+cy.wrap(myAPI)
+-  .invoke('makeARequest', 'http://example.com')
++  .then(api => api.makeARequest('http://example.com'))
+   .then(res => { ...handle response... })
+```
+
+If `.invoke()` is followed by additional commands or assertions, it will call
+the named function multiple times. This has the benefit that the chained
+assertions can more reliably use the function's return value.
+
+If this behavior is undesirable because you expect the function to invoked only
+once, break the command chain and move the chains commands and/or assertions to
+their own chain. For example, rewrite
+
+```diff
+- cy.get('input').invoke('val', 'text').type('newText')
++ cy.get('input').invoke('val', 'text')
++ cy.get('input').type('newText')
+```
+
+#### `.should()`
+
+[`.should()`](/api/commands/should) now throws an error if Cypress commands are
+invoked from inside a `.should()` callback. This previously resulted in unusual
+and undefined behavior. If you wish to execute series of commands on the yield
+value, use`.then()` instead.
+
+```diff
+cy.get('button')
+-  .should(($button) => {
+
+    })
++  .then(api => api.makeARequest('http://example.com'))
+   .then(res => { ...handle response... })
+```
+
+#### `Cypress.Commands.override()`
```

Comment:
```suggestion
#### `Cypress.Commands.overwrite()`
```

Reply:
Courtesy of one of our ambassadors who pointed this out, I would never have noticed myself. :)

Reply:
They should have commented! 😄 I'll update. Thanks for pointing this out.